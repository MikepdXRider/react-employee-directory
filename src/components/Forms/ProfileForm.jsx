import React from 'react'

export default function ProfileForm({user, nameInput, setNameInput, birthdayInput, setBirthdayInput, bioInput, setBioInput, handleSubmit}) {
    
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="name">Name</label>
            <input type="text" id='name' value={nameInput} onChange={(e) => setNameInput(e.target.value)} />

            <label htmlFor="email">Email</label>
            <input type="text" id='email' value={user.email} disabled />

            <label htmlFor="birthday">Birthday</label>
            <input type="date" id='birthday' value={birthdayInput} onChange={(e) => setBirthdayInput(e.target.value)} />

            <label htmlFor="bio">Tell us about yourself</label>
            <textarea id='bio' value={bioInput} onChange={(e) => setBioInput(e.target.value)} />

            <input aria-label='submit form button' type="submit" />
        </form>
    )
}
