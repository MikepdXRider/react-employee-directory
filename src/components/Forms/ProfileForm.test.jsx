import { render } from '@testing-library/react'
import ProfileForm from './ProfileForm.jsx';


it('renders ProfileForm', () => {
    const {container} = render(
            <ProfileForm user={{email: 'test@test.com'}}/>
    )

    expect(container).toMatchSnapshot();
})