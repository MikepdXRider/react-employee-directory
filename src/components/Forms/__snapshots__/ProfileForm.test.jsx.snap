// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProfileForm 1`] = `
<div>
  <form>
    <label
      for="name"
    >
      Name
    </label>
    <input
      id="name"
      type="text"
      value=""
    />
    <label
      for="email"
    >
      Email
    </label>
    <input
      disabled=""
      id="email"
      type="text"
      value="test@test.com"
    />
    <label
      for="birthday"
    >
      Birthday
    </label>
    <input
      id="birthday"
      type="date"
      value=""
    />
    <label
      for="bio"
    >
      Tell us about yourself
    </label>
    <textarea
      id="bio"
    />
    <button
      class="formButton"
    >
      Submit
    </button>
  </form>
</div>
`;
